### DISCUSSION TITLE
[V5] Invalidating cookies, possible?

### DISCUSSION THREAD

**Discussion by aayush123** [adonisjs/core#2644] [Help] [answered]
In my backend, I have two groups of routes:

1. User APIs under the group `api.example.com/api/v1` using OAT Auth
2. Admin APIs under the group `api.example.com/api/v1/admin` using Cookies & Session

Once a user changes their password, I would like to invalidate all other user sessions / OAT.
For OAT, it's pretty straightforward, I simply delete all the tokens issued so far.

But I wonder if it's possible to invalidate all the session cookies issued to a user so far too?
From a security point of view, it would be helpful in cases where your password is compromised, and you change it to secure your account again. It should then also invalidate all ongoing sessions to prevent any further malicious activities, right?

Thanks in advance!

**Comment by thetutlage** [comment, marked as answer]
The cookies are per client (ie browser) and not per user. So you can only clear the session cookie for the current client using `session.clear` method https://docs.adonisjs.com/guides/session#clear

If you have to build a system where all logins will be invalidated, then you will have to building some sort of a custom tracking system that stores the user logins with a key and then drops that key into the cookie as well and then during every request you will validate the key with the database value

**Comment by aayush123** [comment]
Yeah, I was thinking about using a middleware to do all that.
~~On that note, how can I add custom data to the session cookie created by @adonis/session?~~
Using session from context should help with that

**Reply by thetutlage** [reply]
Its covered here in the docs https://docs.adonisjs.com/guides/session#put
